flink-table/flink-table-planner/src/test/java/org/apache/flink/table/planner/plan/nodes/exec/serde/ExecNodeMultiJoinJsonSerdeTest.java

```diff
@@ -0,0 +1,181 @@
+
+class ExecNodeMultiJoinJsonSerdeTest {
+    private final ClassLoader classLoader = Thread.currentThread().getContextClassLoader();
+    private final SerdeContext serdeContext = JsonSerdeTestUtil.configuredSerdeContext();
+
+    @Test
+    void testSerializingStreamExecMultiJoin() throws IOException {
+        // Create test data
+        final StreamExecMultiJoin execNode = createTestMultiJoinNode();
+        final ExecNodeGraph graph =
+                new ExecNodeGraph(FlinkVersion.v2_1, Collections.singletonList(execNode));
+
+        // Test if we can serialize
+        final String serializedGraph = JsonSerdeTestUtil.toJson(serdeContext, graph);
+        assertThat(serializedGraph).isNotEmpty();
+
+        // Test if we can deserialize
+        final ExecNodeGraph deserializedGraph =
+                JsonSerdeTestUtil.toObject(serdeContext, serializedGraph, ExecNodeGraph.class);
+
+        // Verify some general value checks on the deserialized node
+        assertThat(deserializedGraph.getRootNodes()).hasSize(1);
+        final StreamExecMultiJoin deserializedMultiJoinNode =
+                (StreamExecMultiJoin) deserializedGraph.getRootNodes().get(0);
+        assertThat(deserializedMultiJoinNode.getDescription()).isEqualTo("test-multi-join");
+        assertThat(deserializedMultiJoinNode.getOutputType())
+                .isEqualTo(RowType.of(VarCharType.STRING_TYPE, new IntType()));
+    }
+
+    @Test
+    void testSerializedJsonStructure() throws IOException {
+        // Create test data
+        final StreamExecMultiJoin execNode = createTestMultiJoinNode();
+        final ExecNodeGraph graph =
+                new ExecNodeGraph(FlinkVersion.v2_1, Collections.singletonList(execNode));
+
+        // Serialize to JSON
+        final String json = JsonSerdeTestUtil.toJson(serdeContext, graph);
+        final JsonNode jsonNode = new ObjectMapper().readTree(json);
+
+        // Verify JSON structure using JsonSerdeTestUtil assertions
+        // Basic node structure
+        JsonSerdeTestUtil.assertThatJsonContains(jsonNode, "nodes", "0", "type");
+        JsonSerdeTestUtil.assertThatJsonContains(jsonNode, "nodes", "0", "id");
+        JsonSerdeTestUtil.assertThatJsonContains(jsonNode, "nodes", "0", "description");
+        JsonSerdeTestUtil.assertThatJsonContains(jsonNode, "nodes", "0", "inputProperties");
+        JsonSerdeTestUtil.assertThatJsonContains(jsonNode, "nodes", "0", "outputType");
+
+        // MultiJoin specific fields
+        JsonSerdeTestUtil.assertThatJsonContains(jsonNode, "nodes", "0", "joinTypes");
+        JsonSerdeTestUtil.assertThatJsonContains(jsonNode, "nodes", "0", "joinAttributeMap");
+        JsonSerdeTestUtil.assertThatJsonContains(jsonNode, "nodes", "0", "inputUpsertKeys");
+        JsonSerdeTestUtil.assertThatJsonContains(jsonNode, "nodes", "0", "joinConditions");
+        JsonSerdeTestUtil.assertThatJsonContains(jsonNode, "nodes", "0", "stateMetadataList");
```

Comment:
how about dedicated test comparing a compiled plan instead?

Reply:
We can't generate a compiled plan yet, right? We just have the exec node for now and I wanted to somehow test it. But we'll definitely have these as soon as can generate a compiled plan